test/soap.test.js

```diff
@@ -195,6 +195,134 @@ describe('loopback:soap tests', function() {
+            expect(content.properties.myParams.type[0]).to.eql('string');
+
+            content = readModelJsonSync('some-data');
+            expect(content).to.not.have.property('public');
+            expect(content).to.have.property('properties');
+            expect(content.properties.integer_value.type).to.eql('number');
+            expect(content.properties.string_value.type).to.eql('string');
+            expect(content.properties.date_value.type).to.eql('string');
+
+            content = readModelJsonSync('get-some-data-response');
+            expect(content).to.not.have.property('public');
+            expect(content).to.have.property('properties');
+            expect(content.properties.getSomeDataReturn.type).to.be.an('array');
+            expect(content.properties.getSomeDataReturn.type[0]).
+              to.eql('SomeData');
+
+            var modelConfig = readModelConfigSync('server');
+            expect(modelConfig).to.have.property('SomeData');
+            expect(modelConfig.SomeData).to.have.property('public', true);
+            expect(modelConfig).to.have.property('getSomeData');
+            expect(modelConfig.getSomeData).to.have.property('public', true);
+            expect(modelConfig).to.have.property('getSomeDataResponse');
+            expect(modelConfig.getSomeDataResponse).
+              to.have.property('public', true);
+          });
+      });
+  });
+
+  describe('DataService wsdl with model array', function() {
+    beforeEach(common.resetWorkspace);
+    beforeEach(function createSandbox(done) {
+      helpers.testDirectory(SANDBOX, done);
+    });
+    beforeEach(function createProject(done) {
+      common.createDummyProject(SANDBOX, 'test-soapapp', done);
+    });
+    beforeEach(function createDataSource() {
+      return helpers.run(path.join(__dirname, '../datasource'))
+        .cd(SANDBOX)
+        .withPrompts({
+          name: 'ds',
+          connector: 'soap',
+          url: 'http://localhost:8080/DomSoap4/services/DataService',
+          wsdl: path.join(__dirname, 'soap/DataService_model_array.wsdl'),
+          remotingEnabled: true,
+          installConnector: false,
+        }).then();
+    });
+
+    it('DataService wsdl with model array',
+      function() {
+        return helpers.run(path.join(__dirname, '../soap'))
+          .cd(SANDBOX)
+          .withPrompts({
+            dataSource: 'ds',
+            service: 'DataServiceService',
+            binding: 'DataServiceSoapBinding',
+            operations: ['getSomeData'],
+          }).then(function() {
+            var content = readModelJsonSync('get-some-data');
+            expect(content).to.not.have.property('public');
+            expect(content).to.have.property('properties');
+            expect(content.properties.myParams.type).to.be.an('array');
+            expect(content.properties.myParams.type[0]).to.eql('SomeInputData');
+
+            content = readModelJsonSync('some-data');
```

Comment:
just wondering... if the test is only for arrays, do we need to add the additional checks?

Reply:
@dhmlau ,there are previous tests which handle non-arrays and everything else they historically tested. Since my fix `adds` array-handling code, I've added tests for array situations. The other historical tests still handle non-array situations and ensure I haven't broken anything.